src/nap.js

```diff
@@ -157,7 +157,9 @@ function bySelector(){
 }
 
 function wrap(fn, stack) {
-  return stack.reduce(middleware, fn)
+  return function wrapped() {
+    stack.reduce(middleware, fn).apply(null, arguments)
+  }
```

Comment:
It's taken me a long time to figure out how this doesn't actually just break everything, since the return value would be different. I see now that it *isn't*, because the return value before was also a function, albeit the pre-wrapped function. What this does is effectively turn the eager wrapping lazy, such that the middleware is always current and actual. Clever change, but difficult code to follow. Not your fault, but @sammyt's for sure. ;o)

(I kid, I kid! It's brilliant, but I'm stupid.)

Ok, I understand this change now. I'll make the release so we can test and proceed. For what it's worth, I might just produce a word bomb on why I think middleware doesn't make any sense anyway – at least, in this incarnation I don't think they make sense – but I'll leave that in a different issue.

Thanks for your patience @Poetro, I'm not as smart as you so sometimes I need a bit longer, and a couple of walks, to understand things. :o)